src/main/java/emu/grasscutter/server/packet/recv/HandlerPlayerQuitDungeonReq.java

```diff
@@ -1,15 +1,27 @@
 package emu.grasscutter.server.packet.recv;
 
+import emu.grasscutter.game.props.ActivityType;
+import emu.grasscutter.game.activity.trialavatar.TrialAvatarActivityHandler;
 import emu.grasscutter.net.packet.Opcodes;
 import emu.grasscutter.net.packet.PacketOpcodes;
 import emu.grasscutter.net.packet.PacketHandler;
 import emu.grasscutter.server.game.GameSession;
 
+import lombok.val;
+
 @Opcodes(PacketOpcodes.PlayerQuitDungeonReq)
 public class HandlerPlayerQuitDungeonReq extends PacketHandler {
 
     @Override
     public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
+         val playerData = session.getPlayer().getActivityManager()
+            .getPlayerActivityDataByActivityType(ActivityType.NEW_ACTIVITY_TRIAL_AVATAR);
+        if (playerData != null) {
```

Comment:
This is an optional so it needs to be check with `playerData.isPresent()` instead of null.
There is also the getActivityHandler and getActivityHandlerAs methods to directly get the activity handler.
```java

        val playerData = session.getPlayer().getActivityManager().getActivityHandlerAs(ActivityType.NEW_ACTIVITY_TRIAL_AVATAR, TrialAvatarActivityHandler.class);
        if (playerData.isPresent()) {
            val handler = playerData.get();
            handler.unsetTrialAvatarTeam(session.getPlayer());
        }
```

but if you move the dungeon trial character handling to the dungeon manager, it wouldn't even be needed here.